### ISSUE TITLE
Classmethod properties in Python 3.9

### ISSUE THREAD

**Issue by kosayoda** [microsoft/pyright#1423]
**Describe the bug**
In Python 3.9, it's now [possible to chain decorators with `@classmethod`](https://docs.python.org/3/howto/descriptor.html#class-methods). However, pyright raises an error when running the example code given in the Python docs.

**To Reproduce**
Code:
```python
class G:
    @classmethod
    @property
    def __doc__(cls) -> str:
        return f"A doc for {cls.__name__!r}"


print(G.__doc__)
```
Error:
```
3:5 - error: Static and class methods not allowed for property getter, setter or deleter (reportGeneralTypeIssues)
2:6 - error: Argument of type "property" cannot be assigned to parameter "f" of type "(*args: Any, **kwargs: Any) -> Any" in function "__init__"
Type "property" cannot be assigned to type "(*args: Any, **kwargs: Any) -> Any" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos
```
**Expected behavior**
No errors.

**VS Code extension or command-line**
Command-line tool, version `pyright 1.1.104`.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.107, which I just published. It will also be included in the next release of Pylance.
